### DISCUSSION TITLE
Budibase automation (update row) creates five rows

### DISCUSSION THREAD

**Discussion by chiptech-mitchell** [Budibase/budibase#9273] [Help] [answered]
I am trying to update a row created by a Budibase form using the automation, specifically I am connecting to an external data connector (postgres) and attempting to update the row created to have the value returned. 

My automation flow is 
- Row creation trigger
- If condition
- External data connector
- Update row

However even though the automation succeeds the value isn't update into the row, also there are 5 rows created instead of one.

External Data Connector result 
```{
  "response": [
    {
      "id": 45
    }
  ],
```

<img width="571" alt="chrome_zqCuW3bla0" src="https://user-images.githubusercontent.com/109489679/211220367-e90dd82b-8994-4016-b6d1-9d0d99a3a3a6.png">
<img width="370" alt="chrome_tmtrCdlwKt" src="https://user-images.githubusercontent.com/109489679/211220371-171f6076-9cae-4736-9c8b-a1ac6e36a6b2.png">
<img width="227" alt="chrome_FeUCtJyKPc" src="https://user-images.githubusercontent.com/109489679/211220370-32bbc7a3-5683-43a5-9568-8fae8cf8a964.png">

If anyone can help that would be appreciated.

**Comment by melohagan** [comment]
Hey @chiptech-mitchell 

That's odd, would you be able to share your app export to support@budibase.com please? Then I can take a closer look.

**Reply by chiptech-mitchell** [reply]
Unfortunately there is stuff in the app that I am unable to share, is there away to duplicate the application I can try see if I can remove anyway unneeded information and have a minimum reproducible app

**Reply by melohagan** [reply]
That would be great, you can export an app without the rows by toggling the *exclude rows* switch, and then re-import it as a new app.
Then in your new app you can remove any remaining sensitive info and export that.

**Reply by chiptech-mitchell** [reply]
I am unable to export the application as I get a 502 every time I try, and I am having shelve my work on Budibase for the moment

**Reply by melohagan** [reply]
Hey @chiptech-mitchell are you on cloud hosting? If you are can you provide your tenant id?

**Reply by chiptech-mitchell** [reply]
No, we are using self hosting

**Comment by melohagan** [comment, marked as answer]
Hey @chiptech-mitchell I just had a thought - Is the Row creation trigger for the same table as your Update row action?
*Edit: I see that it is, the Customer table*

This could be causing an infinite loop (although Budibase forcibly breaks after 5 iterations). 

As for the Row ID, the binding should really be: `{{ trigger.row._id }}`. Hope that helps!

**Reply by chiptech-mitchell** [reply]
Oh well that makes sense I was retriggering the automation infinitely, Yes I was trying to update a field that I wasn't able to get a value in before saving the entry. We are probably going to use an api to do the heavy lifting for customer creation at this stage.

Thanks for helping out.
